skops/card/_model_card.py

```diff
@@ -1,61 +1,120 @@
-    card_data.library_name = "sklearn"
-    template_path = card_kwargs.get("template_path")
-    if template_path is None:
-        template_path = os.path.join(f"{ROOT[0]}", "card", "default_template.md")
-    card_kwargs["template_path"] = template_path
-    card = ModelCard.from_template(
-        card_data=card_data,
-        hyperparameter_table=hyperparameter_table,
-        model_plot=model_plot,
-        **card_kwargs,
-    )
-
-    return card
+class Card:
+    def __init__(self, model):
+        self.model = model
+        self.hyperparameter_table = self._extract_estimator_config()
+        # the spaces in the pipeline breaks markdown, so we replace them
+        self.model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
+        self.template_sections = {}
+        self.figure_paths = {}
+
+    def add(self, **kwargs):
+        """Takes values to fill model card template.
+        Parameters:
+        ----------
+        **kwargs : dict
+            Parameters to be set for the model card. These parameters
+            need to be sections of the underlying `jinja` template used.
+
+        """
+        for section, value in kwargs.items():
+            self.template_sections[section] = value
+        return self
+
+    def add_inspection(self, **kwargs):
+        """Takes plots to fill model card template.
+        Parameters:
+        ----------
+        Parameters to be set for the model card. These parameters
+        need to be sections of the underlying `jinja` template used.
+        """
+        for plot_name, plot_path in kwargs.items():
+            self.figure_paths[plot_name] = plot_path
+        return self
+
+    def save(self, path):
+        """Fills model card template, renders and saves it as markdown file to the target directory.
+
+        Parameters:
+        ----------
+        path: Path
+              filepath to save your card.
+        """
+        ROOT = skops.__path__
+
+        metadata_keys = [
+            "language",
+            "license",
+            "library_name",
+            "tags",
+            "datasets",
+            "model_name",
+            "metrics",
+            "model-index",
+        ]
+        card_data_keys = {}
+
+        # if key is supposed to be in metadata and is provided by user, write it to card_data_keys
+        for key in metadata_keys:
+            if key in self.template_sections.keys():
+                card_data_keys[key] = self.template_sections.pop(key, "")
+
+        # construct CardData
+        card_data = CardData(**card_data_keys)
+        card_data.library_name = "sklearn"
+
+        if self.template_sections.get("template_path") is None:
+            self.template_sections["template_path"] = os.path.join(
+                f"{ROOT[0]}", "card", "default_template.md"
+            )
+
+        # append plot_name if any plots are provided, at the end of the template
+        if self.figure_paths != {}:
+            shutil.copyfile(
```

Comment:
Sorry, I don't quite understand what is going on here. Could you please add a comment to explain why these steps are necessary?

Reply:
sure. basically we can't edit jinja template unless we add the slots in the template, that's why I had to append at the end.